extensions/wikia/AdEngine/js/spec/config.desktop.spec.js

```diff
@@ -0,0 +1,188 @@
+/*global describe, it, expect, modules, spyOn*/
+describe('ext.wikia.adEngine.config.desktop', function () {
+	'use strict';
+
+	function noop() {
+		return;
+	}
+
+	function returnEmpty() {
+		return {};
+	}
+
+	var uaIE8 = [
+			'Mozilla/5.0 (compatible; MSIE 8.0; Windows NT 6.1; Trident/4.0;',
+			'GTB7.4; InfoPath.2; SV1; .NET CLR 3.3.69573; WOW64; en-US)'
+		].join(''),
+		mocks = {
+			adDecoratorPageDimensions: noop,
+			getAdContextOpts: returnEmpty,
+			getAdContextTargeting: returnEmpty,
+			getAdContextProviders: returnEmpty,
+			getAdContextForceProviders: returnEmpty,
+			getInstantGlobals: returnEmpty,
+			getUserAgent: noop,
+			geo: {
+				getCountryCode: noop
+			},
+			log: noop,
+			providers: {
+				directGpt: {
+					name: 'direct'
+				},
+				evolve: {
+					name: 'evolve',
+					canHandleSlot: noop
+				},
+				liftium: {
+					name: 'liftium'
+				},
+				remnantGpt: {
+					name: 'remnant'
+				},
+				sevenOneMedia: {
+					name: 'sevenOneMedia'
+				},
+				taboola: {
+					name: 'taboola',
+					canHandleSlot: noop
+				},
+				turtle: {
+					name: 'turtle'
+				}
+			}
+		};
+
+	function getModule() {
+		return modules['ext.wikia.adEngine.config.desktop'](
+			mocks.log,
+			{navigator: {userAgent: mocks.getUserAgent()}},
+			mocks.getInstantGlobals(),
+			mocks.geo,
+			{
+				getContext: function () {
+					return {
+						opts: mocks.getAdContextOpts(),
+						targeting: mocks.getAdContextTargeting(),
+						providers: mocks.getAdContextProviders(),
+						forceProviders: mocks.getAdContextForceProviders()
+					};
+				}
+			},
+			mocks.adDecoratorPageDimensions,
+			mocks.providers.evolve,
+			mocks.providers.liftium,
+			mocks.providers.directGpt,
+			mocks.providers.remnantGpt,
+			mocks.providers.sevenOneMedia,
+			mocks.providers.turtle,
+			mocks.providers.taboola
+		);
+	}
+
+	function getProviders(slotName) {
+		var providers = getModule().getProviderList(slotName),
+			providerNames = providers.map(function (item) {
+				return item.name;
+			});
+
+		return providerNames.join(',');
+	}
+
+	it('default setup: Direct, Remnant, Liftium', function () {
+		expect(getProviders('foo')).toEqual('direct,remnant,liftium');
+	});
+
+	it('non-Evolve country, Evolve slot: Direct, Remnant, Liftium', function () {
+		expect(getProviders('foo')).toEqual('direct,remnant,liftium');
+	});
+
+	it('NZ, Evolve slot: Evolve, Remnant, Liftium', function () {
+		spyOn(mocks.geo, 'getCountryCode').and.returnValue('NZ');
+		spyOn(mocks.providers.evolve, 'canHandleSlot').and.returnValue(true);
+		expect(getProviders('foo')).toEqual('evolve,remnant,liftium');
+	});
+
+	it('NZ, not Evolve slot: Direct, Remnant, Liftium', function () {
+		spyOn(mocks.geo, 'getCountryCode').and.returnValue('NZ');
```

Comment:
We don't spy on `mocks.providers.evolve`'s `canHandleSlot()` method because "by default" it's `noop()` call - is it correct assumption? I wonder if it won't be better to explicitly "show" that the results depends mostly on evolve provider's `canHandleSlot()` result. So, my proposal is to add line:
```js
spyOn(mocks.providers.evolve, 'canHandleSlot').and.returnValue(false);
```